### PR TITLE
fix(multigrid): mouse events crash neovim

### LINKED ISSUE #15075: multigrid: mouse events crash neovim
- `nvim --version`: NVIM v0.6.0-dev+32-ge43dbfdd0
- Operating system/version: win10-x64 build 19042
- UI: fvim/neovide (with multigrid support)

### Steps to reproduce using `nvim -u NORC`

```
fvim --debug-multigrid -u NORC
- or -
neovide --multiGrid -u NORC


:set mouse=a
press down mouse from within the buffer grid (usually grid #2)
drag mouse down towards status line (part of the "base grid #1")
once the mouse hits the status line, neovim crashes.

```

https://user-images.githubusercontent.com/20684720/125405892-cb718900-e3ea-11eb-9920-49f9c3f8878f.mp4

### PR DESCRIPTION
**Pull request by yatli**
fix #15075

### CODE REVIEW

File: src/nvim/mouse.c [hunk 1]
```diff
@@ -721,14 +721,20 @@ int mouse_check_fold(void)
...
+  if (wp && multigrid) {
+    max_row = wp->w_grid_alloc.Rows;
+    max_col = wp->w_grid_alloc.Columns;
+  }
 
-  if (wp && mouse_row >= 0 && mouse_row < Rows
-      && mouse_col >= 0 && mouse_col <= Columns) {
-    int multigrid = ui_has(kUIMultigrid);
+  if (wp && mouse_row >= 0 && mouse_row < max_row
+      && mouse_col >= 0 && mouse_col <= max_col) {
```
Comment (yatli) [inline src/nvim/mouse.c:737]: `mouse_col <= max_col` or `mouse_col < max_col`? 🤔

Comment (bfredl) [reply src/nvim/mouse.c:737]: probably the latter. rows and colums should work the same.

Comment (yatli) [reply src/nvim/mouse.c]: ok fixed

**Comment by bfredl** [comment]
a test would be good as well :]

**Comment by yatli** [comment]
@bfredl the test does not crash the base version, but indeed produces incorrect result.

**Comment by bfredl** [comment]
Yes, that's good enough.

**Comment by yatli** [comment]
btw, what does these numbers in the grid mean?
```
      {1:~                                               }|
    ## grid 3
      {7:-- VISUAL --}                                    |
```
hlid? I see different numbers in mouse tests.

**Comment by bfredl** [comment]
yes, those are defined in the `set_default_attr_ids` table.

**Comment by yatli** [comment]
ringring @bfredl

**Comment by bfredl** [comment]
unrelated error, trying a restart.

**Comment by yatli** [comment]
@bfredl can we run the tests again -- got a (unrelated?) segfault `"test49.ok" "test49.ok" 56 lines, 1748 charactersE484: Can't open file C:/Program Files/nvim/share/nvim\syntax\syntax.vimrunnvim.sh: line 91:   179 Segmentation fault      "$nvim_prg" -u NONE -i NONE --headless --cmd 'set shortmess+=I noswapfile noundofile nomore' -S runnvim.vim "$tlog" > "out-$tlog" 2> "err-$tlog"`

**Pull request merged**